pkg/test/context.go

```diff
@@ -0,0 +1,97 @@
+// Copyright 2018 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package test
+
+import (
+	"log"
+	"strconv"
+	"strings"
+	"testing"
+	"time"
+
+	"k8s.io/client-go/rest"
+)
+
+type TestCtx struct {
+	ID         string
+	CleanUpFns []finalizerFn
+	Namespace  string
+	CRClient   *rest.RESTClient
+}
+
+type finalizerFn func() error
+
+func (f *Framework) NewTestCtx(t *testing.T) TestCtx {
```

Comment:
is there a reason to tie `NewTestCtx` to `Framework` object? it seems me that `NewTestCtx` can just be a static factory function.